lib/signed-xml.js

```diff
@@ -367,19 +418,13 @@ SignedXml.prototype.checkSignature = function (xml, callback) {
   this.validationErrors = [];
   this.signedXml = xml;
 
-  if (!this.keyInfoProvider) {
-    var err = new Error("cannot validate signature since no key info resolver was provided");
-    if (!callback) {
-      throw err;
-    } else {
-      callback(err);
-      return;
-    }
-  }
-
-  this.signingKey = this.keyInfoProvider.getKey(this.keyInfo);
-  if (!this.signingKey) {
-    var err2 = new Error("key info provider could not resolve key info " + this.keyInfo);
+  const signingCert = this.getCertFromKeyInfo(this.keyInfo) || this.signingCert;
+  if (!signingCert) {
+    var err2 = new Error(
+      "key info provider could not resolve key info " +
+        this.keyInfo +
+        "\n and no publicCert provided"
+    );
```

Comment:
This should probably not throw an error if `signingKey `is set. `validateSignatureValue `will also work with only the signingKey (think about symmetric algorithms)